tools/json2bin/evo.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python3
+import pathlib
+
+
+import json2bin as j2b
+
+
+from consts import (
+    item,
+    moves
+)
+
+
+from consts.pokemon import (
+    evo_methods,
+    species
+)
+
+
+def get_evo_params(method: evo_methods.EvoMethod, evo: list):
+    maybe_param = evo[1]
+    final_param = 0
+    #None of these take an extra parameter
+    if method in set([
+        evo_methods.EvoMethod.EVO_NONE,
+        evo_methods.EvoMethod.EVO_HAPPINESS,
+        evo_methods.EvoMethod.EVO_HAPPINESS_DAY,
+        evo_methods.EvoMethod.EVO_HAPPINESS_NIGHT,
+        evo_methods.EvoMethod.EVO_TRADE,
+        evo_methods.EvoMethod.EVO_ELECTRIC_FIELD,
+        evo_methods.EvoMethod.EVO_MOSSY_STONE,
+        evo_methods.EvoMethod.EVO_ICY_STONE]):
+        final_param = 0
+    # These all specify a basic integer param
+    elif method in set([
+        evo_methods.EvoMethod.EVO_LEVEL_UP,
+        evo_methods.EvoMethod.EVO_LEVEL_ATK_GT_DEF,
+        evo_methods.EvoMethod.EVO_LEVEL_ATK_EQ_DEF,
+        evo_methods.EvoMethod.EVO_LEVEL_ATK_LT_DEF,
+        evo_methods.EvoMethod.EVO_LEVEL_PID_LOW,
+        evo_methods.EvoMethod.EVO_LEVEL_PID_HIGH,
+        evo_methods.EvoMethod.EVO_LEVEL_NINJASK,
+        evo_methods.EvoMethod.EVO_LEVEL_SHEDINJA,
+        evo_methods.EvoMethod.EVO_LEVEL_MALE,
+        evo_methods.EvoMethod.EVO_LEVEL_FEMALE,
+        evo_methods.EvoMethod.EVO_BEAUTY]):
+        final_param = maybe_param
+    # These specify an item
+    elif method in set([
+        evo_methods.EvoMethod.EVO_TRADE_WITH_ITEM,
+        evo_methods.EvoMethod.EVO_USE_ITEM,
+        evo_methods.EvoMethod.EVO_USE_ITEM_MALE,
+        evo_methods.EvoMethod.EVO_USE_ITEM_FEMALE,
+        evo_methods.EvoMethod.EVO_USE_ITEM_DAY,
+        evo_methods.EvoMethod.EVO_USE_ITEM_NIGHT]):
+        final_param = item.Item[maybe_param].value
+    elif method == evo_methods.EvoMethod.EVO_KNOW_MOVE:
+        final_param = moves.Move['MOVE_' + maybe_param].value
+    elif method == evo_methods.EvoMethod.EVO_MON_IN_PARTY:
+        final_param = species.PokemonSpecies[maybe_param].value
+
+    return final_param
+
+
+def table_line(evo_method: int, evo_params: int, species: int) -> bytes:
+    binary = bytearray([])
+    binary.extend((evo_method & 0x7F).to_bytes(2, 'little'))
+    binary.extend((evo_params & 0x7F).to_bytes(2, 'little'))
+    binary.extend((species & 0x7F).to_bytes(2, 'little'))
+    return bytes(binary)
+    
+
+def parse_evolutions(table: list, _size: int, _enum: None):
+    out = []
+    for j in range(min(len(table), 7)):
+        evo = table[j]
+        method = evo_methods.EvoMethod['EVO_' + evo[0]]
```

Comment:
Let's update the data files themselves to have this prefix automatically. I can write up a shell script for you to accomplish that, if you'd like! 😁

Reply:
I should be ablet to that myself. I just wanted to get the script working properly first.

Reply:
I've updated the 'evolutions' entries to have the evolution methods be 'EVO_' prepended and same thing for the moves if a pokemon evolved by knowing a move (the moves were not prepended with 'MOVE_' before)